src/Microsoft.Health.Dicom.Core/Features/Store/StoreDatasetValidator.cs

```diff
@@ -87,9 +87,9 @@ private static void ValidateCoreTags(DicomDataset dicomDataset, string requiredS
                 DicomCoreResource.DuplicatedUidsNotAllowed);
         }
 
-        // If the requestedStudyInstanceUid is specified, then the StudyInstanceUid must match.
+        // If the requestedStudyInstanceUid is specified, then the StudyInstanceUid must match, ignoring whitespace.
         if (requiredStudyInstanceUid != null &&
-            !studyInstanceUid.Equals(requiredStudyInstanceUid, StringComparison.OrdinalIgnoreCase))
+            !studyInstanceUid.Trim().Equals(requiredStudyInstanceUid.Trim(), StringComparison.OrdinalIgnoreCase))
```

Comment:
one note here - we only want to trim trailing whitespace since that's the SQL behavior.